src/Mvc/Mvc.RazorPages/src/CompiledPageActionDescriptor.cs

```diff
@@ -65,5 +65,7 @@ public CompiledPageActionDescriptor(PageActionDescriptor actionDescriptor)
         /// Gets or sets the associated <see cref="Endpoint"/> of this page.
         /// </summary>
         public Endpoint Endpoint { get; set; }
+
+        internal PageActionInvokerCacheEntry PageActionInvokerCacheEntry { get; set; }
```

Comment:
@rynowak how gross is this? Having this here avoids a second dictionary lookup per page invocation.

Reply:
Action Descriptors are data. They are intentionally constructable and don't have "initialized" state on them. It's important that we can construct valid ADs without needing services.